backend/services/grading_embedding_service.py

```
@@ -0,0 +1,388 @@
+                    )[0][0]
+                    chunk_similarities.append({
+                        'chunk': chunk,
+                        'similarity': similarity
+                    })
+            
+            # Sort by similarity and return top chunks
+            chunk_similarities.sort(key=lambda x: x['similarity'], reverse=True)
+            top_chunks = [item['chunk'] for item in chunk_similarities[:max_chunks]]
+            
+            debug_log(f"[GRADING_RETRIEVAL] Retrieved {len(top_chunks)} relevant chunks")
+            return top_chunks
```

Ruff (0.14.1)

Consider moving this statement to an `else` block

(TRY300)